Pt 2289 allow pt stalk do disable ps lock transactions data collection via parameter

svetasmirnova opened the pull request:
- [x] The contributed code is licensed under GPL v2.0
- [x] Contributor Licence Agreement (CLA) is signed
- [x] util/update-modules has been ran
- [x] Documentation updated
- [x] Test suite update

svetasmirnova committed:
PT-2289 - Allow pt-stalk do disable ps-lock-transactions data collection via parameter

- Implemented feature
- Started working on the test

svetasmirnova committed:
PT-2289 - Allow pt-stalk do disable ps-lock-transactions data collection via parameter

- Adjusted the implementation
- Created test cases

EvgeniyPatlan commented on bin/pt-stalk:
```diff
@@ -971,8 +971,10 @@ collect_mysql_data_one() {
 
-   $CMD_MYSQLADMIN $EXT_ARGV ext -i$OPT_SLEEP_COLLECT -c$cnt >>"$d/$p-mysqladmin" &
-   mysqladmin_pid=$!
+   if ! [[ "${OPT_SKIP_COLLECTION[@]}" =~ "mysqladmin" ]]; then
```
Bash does not support regex matching directly against arrays using =~.

use something like this
```
should_skip=false
for item in "${OPT_SKIP_COLLECTION[@]}"; do
    if [[ "$item" == "mysqladmin" ]]; then
        should_skip=true
        break
    fi
done

if ! $should_skip; then
    "$CMD_MYSQLADMIN" $EXT_ARGV ext -i"$OPT_SLEEP_COLLECT" -c"$cnt" >> "$d/$p-mysqladmin" &
    mysqladmin_pid=$!
fi
```

svetasmirnova replied:
Fixed, please review again

EvgeniyPatlan commented on bin/pt-stalk:
```diff
@@ -1014,18 +1016,25 @@ collect_system_data() {
 collect_mysql_data_loop() {
 
-
+   if ! [[ "${OPT_SKIP_COLLECTION[@]}" =~ "processlist" ]]; then
```
the same

EvgeniyPatlan commented on bin/pt-stalk:
```diff
@@ -1014,18 +1016,25 @@ collect_system_data() {
    if [ "$have_lock_waits_table" ]; then
-      (echo $ts; lock_waits "$d/lock_waits.running")   >>"$d/$p-lock-waits" &
-      (echo $ts; transactions) >>"$d/$p-transactions" &
+      if ! [[ "${OPT_SKIP_COLLECTION[@]}" =~ "lock-waits" ]]; then
```
the same

svetasmirnova committed:
PT-2289 - Allow pt-stalk do disable ps-lock-transactions data collection via parameter

- Implemented _should_skip function as was suggested

svetasmirnova committed:
PT-2289 - Allow pt-stalk do disable ps-lock-transactions data collection via parameter

- Modified t/pt-stalk/pt-stalk.t, so it calls function purge_samples with the correct number of parameters
- Fixed purge_samples, so it can work if variable OPT_PREFIX is not defined

EvgeniyPatlan commented on bin/pt-stalk:
```diff
@@ -1356,18 +1367,33 @@ collect_mysql_variables() {
+   local name=$1
+
+   for item in "${OPT_SKIP_COLLECTION[@]}"; do
+   echo $item >> /tmp/sveta
```
let's implement another name here

svetasmirnova replied:
OMG, I need some automation to remove those print debug messages.

svetasmirnova committed:
PT-2289 - Allow pt-stalk do disable ps-lock-transactions data collection via parameter

- Fixed typo found during review
